src/LogFileReader.cpp

```diff
@@ -67,19 +71,38 @@ bool LogFileReader::readAll(const QString& fileName)
             emit batteryDataReceived(batteryData);
         }
     }
-
     return true;
 }
 
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
+    QStringList lines = line.split(COMMA);
+    QTime Time = QTime::fromString(lines[0], TIMEFORMAT);
+    bool ok;
+    if (Time.isValid())
+    {
+        batteryData.time = Time;
+    }
+    else
+    {
+        return false;
+    }
+
+    if (lines.count() !=    3)
+    {
+        return false;
+    }
 
-    // This is here to the compiler happy. Otherwise the compile
-    // will have an error warning about an unused variable. Remove this
-    // when you use it.
-    Q_UNUSED(line);
-    Q_UNUSED(batteryData);
+    for (int i = 1; i<=2; i++)
+    {
+        lines[i].toDouble(&ok);
+        if (!ok)
+            return false;
```

Comment:
All if else statements should be enclosed in braces regardless if they're one liners or not. This is because code quickly becomes not one liners, and also the consistency makes the code easier to read.